app/ide-desktop/lib/dashboard/src/authentication/src/authentication/components/login.tsx

```diff
@@ -1,18 +1,138 @@
+    auth.useAuth();
+
+  const initialEmail = parseUrlSearchParams(search);
+
+  const { value: email, bind: bindEmail } = hooks.useInput(initialEmail ?? "");
+  const { value: password, bind: bindPassword } = hooks.useInput("");
+
   return (
     <div className="min-h-screen flex flex-col items-center justify-center bg-gray-300">
       <div className="flex flex-col bg-white shadow-md px-4 sm:px-6 md:px-8 lg:px-10 py-8 rounded-md w-full max-w-md">
+        <div className="font-medium self-center text-xl sm:text-2xl uppercase text-gray-800">
+          Login To Your Account
+        </div>
+        <button
+          onClick={utils.handleEvent(signInWithGoogle)}
+          className="relative mt-6 border rounded-md py-2 text-sm text-gray-800 bg-gray-100 hover:bg-gray-200"
+        >
+          <span className="absolute left-0 top-0 flex items-center justify-center h-full w-10 text-blue-500">
+            <fontawesome.FontAwesomeIcon icon={fontawesomeIcons.faGoogle} />
+          </span>
+          <span>Login with Google</span>
+        </button>
+        <button
+          onClick={utils.handleEvent(signInWithGitHub)}
+          className="relative mt-6 border rounded-md py-2 text-sm text-gray-800 bg-gray-100 hover:bg-gray-200"
+        >
+          <span className="absolute left-0 top-0 flex items-center justify-center h-full w-10 text-blue-500">
+            <fontawesome.FontAwesomeIcon icon={fontawesomeIcons.faGithub} />
+          </span>
+          <span>Login with Github</span>
+        </button>
+        <div className="relative mt-10 h-px bg-gray-300">
+          <div className="absolute left-0 top-0 flex justify-center w-full -mt-2">
+            <span className="bg-white px-4 text-xs text-gray-500 uppercase">
+              Or Login With Email
+            </span>
+          </div>
+        </div>
+        <div className="mt-10">
+          <form
+            onSubmit={utils.handleEvent(async () =>
+              signInWithPassword(email, password)
+            )}
+          >
+            <div className="flex flex-col mb-6">
+              <label
+                htmlFor="email"
+                className="mb-1 text-xs sm:text-sm tracking-wide text-gray-600"
+              >
+                E-Mail Address:
+              </label>
+              <div className="relative">
+                <div className="inline-flex items-center justify-center absolute left-0 top-0 h-full w-10 text-gray-400">
+                  <span>
+                    <icons.Svg data={icons.PATHS.at} />
+                  </span>
+                </div>
+
+                <input
+                  {...bindEmail}
+                  required={true}
+                  id="email"
+                  type="email"
+                  name="email"
+                  className="text-sm sm:text-base placeholder-gray-500 pl-10 pr-4 rounded-lg border border-gray-400 w-full py-2 focus:outline-none focus:border-blue-400"
```

Comment:
several lines > 100 chars. Please remember about line folding

Reply:
All over-long lines appear to be fixed.